jenkins/version-def.sh

```diff
@@ -48,9 +57,20 @@ SPARK_REPO=${SPARK_REPO:-"$URM_URL"}
 echo "CUDF_VER: $CUDF_VER, CUDA_CLASSIFIER: $CUDA_CLASSIFIER, PROJECT_VER: $PROJECT_VER \
     SPARK_VER: $SPARK_VER, SCALA_BINARY_VER: $SCALA_BINARY_VER"
 
+case $PHASE_TYPE in
 
-SPARK_SHIM_VERSIONS_STR=${SPARK_SHIM_VERSIONS_STR:-"311 321cdh 312 313 314 320 321 322 330 331"}
+    pre-release)
+        PROFILE_OPT="-Psnapshots" 
+        ;;
 
-IFS=" " <<< $SPARK_SHIM_VERSIONS_STR read -r -a SPARK_SHIM_VERSIONS
+    *)
+        PROFILE_OPT="-PnoSnapshots" 
+        ;;
+esac
 
-SPARK_BASE_SHIM_VERSION=${SPARK_SHIM_VERSIONS[0]}
+SPARK_SHIM_VERSIONS_STR=$(mvn -B help:evaluate -q -pl dist $PROFILE_OPT -Dexpression=included_buildvers -DforceStdout)
+SPARK_SHIM_VERSIONS_STR=$(echo $SPARK_SHIM_VERSIONS_STR)
+PRE_IFS=$IFS
+IFS=", " <<< $SPARK_SHIM_VERSIONS_STR read -r -a SPARK_SHIM_VERSIONS
+IFS=$PRE_IFS
```

Comment:
looks like a third instance of the same logic, maybe factor out as a function in another file that is sourced everywhere needed?